### PR TITLE
add tofu_jr keyboard

### PR DESCRIPTION
**Pull request by moyi4681**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/dztech/tofu/jr/v1/halconf.h [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#pragma once
```
Comment (drashna) [inline keyboards/dztech/tofu/jr/v1/halconf.h:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/dztech/tofu/jr/v1/mcuconf.h [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#pragma once
```
Comment (drashna) [inline keyboards/dztech/tofu/jr/v1/mcuconf.h:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/dztech/tofu/jr/v1/v1.c [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+}, {
+    1, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 1,
+    1, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 1,
+    1, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 1, 
+    1, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 1,
+    1, 1, 1, 4, 1, 1, 1, 1, 1, 1	
+} };
+
+__attribute__ ((weak))
+void rgb_matrix_indicators_user(void)
```
Comment (drashna) [inline keyboards/dztech/tofu/jr/v1/v1.c:116]: Note to self: #18448

File: users/mechmerlin/mechmerlin.c [hunk 4]
```diff
@@ -1,6 +1,6 @@
 #include "mechmerlin.h"
 #include "version.h"
-
+#undef TAP_CODE_DELAY
```
Comment (drashna) [inline users/mechmerlin/mechmerlin.c:3]: please revert this change.

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/dztech/tofu/jr/v1/v1.c [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    { 0, K_12, J_12, L_12 },
+    { 0, K_11, J_11, L_11 },
+    { 0, K_10, J_10, L_10 },
+    { 0, K_7,  J_7,  L_7 },
+    { 0, K_4,  J_4,  L_4 },
+    { 0, K_3,  J_3,  L_3 },
+    { 0, K_2,  J_2,  L_2 },
+    { 0, K_1,  J_1,  L_1 },    
+    { 0, K_6,  J_6,  L_6 },
+    { 0, K_5,  J_5,  L_5 },
```
Comment (fauxpark) [inline keyboards/dztech/tofu/jr/v1/v1.c:93]:
```suggestion
    { 0, K_5,  J_5,  L_5 },
```

File: keyboards/dztech/tofu/jr/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    [0] = LAYOUT_65_ansi(
+        KC_ESC,   KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_HOME,
+        KC_TAB,   KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_PGUP,
+        KC_CAPS,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,  KC_PGDN,
+        KC_LSFT,  KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,          KC_UP,   KC_END,
+        KC_LCTL,  KC_LGUI, KC_LALT,                            KC_SPC,           KC_RALT, KC_RGUI, KC_RCTL,          KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+    [1] = LAYOUT_65_ansi(
+        KC_GESC, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,  KC_HOME,
+        _______, RGB_TOG, RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, _______, KC_PSCR, KC_SLCK, KC_PAUS, QK_BOOT,   KC_PGUP,
```
Comment (fauxpark) [inline keyboards/dztech/tofu/jr/keymaps/default/keymap.c:29]:
```suggestion
        _______, RGB_TOG, RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, _______, KC_PSCR, KC_SCRL, KC_PAUS, QK_BOOT, KC_PGUP,
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/dztech/tofu/jr/v1/v1.c [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+ #include "is31fl3737.h"
```
Comment (fauxpark) [inline keyboards/dztech/tofu/jr/v1/v1.c:17]:
```suggestion
 #include "quantum.h"
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/dztech/tofu/jr/v1/v1.c [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+    { 0, K_6,  J_6,  L_6 },
+    { 0, K_5,  J_5,  L_5 },
+};
+
+__attribute__ ((weak))
+
+void rgb_matrix_indicators_user(void)
+{
+    if (host_keyboard_led_state().caps_lock)
+    {
```
Comment (drashna) [inline keyboards/dztech/tofu/jr/v1/v1.c]:
```suggestion
bool rgb_matrix_indicators_kb(void) {
    if (!rgb_matrix_indicators_user()) {
        return false;
    }
    if (host_keyboard_led_state().caps_lock) {
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/dztech/tofu/jr/v1/v1.c [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+
+__attribute__ ((weak))
+bool rgb_matrix_indicators_kb(void) {
+    if (!rgb_matrix_indicators_user()) {
+        return false;
+    }
+    if (host_keyboard_led_state().caps_lock) {
+        rgb_matrix_set_color(30, 0xFF, 0xFF, 0xFF);
+    }
+}
```
Comment (drashna) [inline keyboards/dztech/tofu/jr/v1/v1.c:104]:
```suggestion
    return true;
}
```

File: keyboards/dztech/tofu/jr/v1/v1.c [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+    { 0, K_7,  J_7,  L_7 },
+    { 0, K_4,  J_4,  L_4 },
+    { 0, K_3,  J_3,  L_3 },
+    { 0, K_2,  J_2,  L_2 },
+    { 0, K_1,  J_1,  L_1 },
+    { 0, K_6,  J_6,  L_6 },
+    { 0, K_5,  J_5,  L_5 },
+};
+
+__attribute__ ((weak))
```
Comment (drashna) [inline keyboards/dztech/tofu/jr/v1/v1.c:96]:
```suggestion
```

File: keyboards/dztech/tofu/jr/v1/config.h [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+#    define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
+#    define RGB_MATRIX_MAXIMUM_BRIGHTNESS 180 // limits maximum brightness of LEDs to 200 out of 255. If not defined maximum brightness is set to 255
+#    define RGB_MATRIX_STARTUP_MODE RGB_MATRIX_CYCLE_ALL
+#    define DRIVER_ADDR_1 0b1010000
+#    define DRIVER_ADDR_2 0b1011111
+#    define DRIVER_COUNT 2
+#    define DRIVER_1_LED_TOTAL 48
+#    define DRIVER_2_LED_TOTAL 20
+#    define DRIVER_LED_TOTAL (DRIVER_1_LED_TOTAL + DRIVER_2_LED_TOTAL)
+#    define RGB_MATRIX_LED_COUNT DRIVER_LED_TOTAL
```
Comment (fauxpark) [inline keyboards/dztech/tofu/jr/v1/config.h:88]:
```suggestion
#    define RGB_MATRIX_LED_COUNT (DRIVER_1_LED_TOTAL + DRIVER_2_LED_TOTAL)
```

File: keyboards/dztech/tofu/jr/v1/config.h [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
+#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS
+#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
+#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
+#    define ENABLE_RGB_MATRIX_SPLASH
+#    define ENABLE_RGB_MATRIX_MULTISPLASH
+#    define ENABLE_RGB_MATRIX_SOLID_SPLASH
+#    define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
+#    define RGB_MATRIX_MAXIMUM_BRIGHTNESS 180 // limits maximum brightness of LEDs to 200 out of 255. If not defined maximum brightness is set to 255
+#    define RGB_MATRIX_STARTUP_MODE RGB_MATRIX_CYCLE_ALL
```
Comment (fauxpark) [inline keyboards/dztech/tofu/jr/v1/config.h:81]:
```suggestion
#    define RGB_MATRIX_DEFAULT_MODE RGB_MATRIX_CYCLE_ALL
```

File: keyboards/dztech/tofu/jr/v1/info.json [hunk 13]
```diff
@@ -0,0 +1,177 @@
...
+        "rgb_matrix": true,
+    },
+    "matrix_pins": {
+        "cols": ["GP25", "GP24", "GP23", "GP1", "GP7", "GP21", "GP20", "GP19", "GP18", "GP17", "GP16", "GP15", "GP14", "GP13", "GP12"],
+        "rows": ["GP29", "GP28", "GP27", "GP26", "GP22"]
+    },
+    "matrix_size": {
+        "cols": 15,
+        "rows": 5
+    },
```
Comment (fauxpark) [inline keyboards/dztech/tofu/jr/v1/info.json:22]:
```suggestion
```

File: keyboards/dztech/tofu/jr/v1/v1.c [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+ #include "quantum.h"
+
+#ifdef RGB_MATRIX_ENABLE
+
+const is31_led PROGMEM g_is31_leds[DRIVER_LED_TOTAL] = {
```
Comment (fauxpark) [inline keyboards/dztech/tofu/jr/v1/v1.c:21]:
```suggestion
const is31_led PROGMEM g_is31_leds[RGB_MATRIX_LED_COUNT] = {
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by fauxpark** [comment]
Last commit should be reverted,I think, and fixed on develop after merge.

**Review by tzarc — APPROVED** [review]
